### DISCUSSION TITLE
NetBox maintainers meeting minutes, 20 June 2024

### DISCUSSION THREAD

**Discussion by jeffgdotorg** [netbox-community/netbox#16673] [Announcements]
*NetBox maintainers meeting minutes, 20 June 2024*

_Attending_: Arthur Hanson, Jeff Gehlbach, Ben Reed, Dan Sheppard, Jeremy Stretch

_[Jeremy] [Bug #14634](https://github.com/netbox-community/netbox/issues/14634)_
- Identifying a workable solution has been challenging (thanks Dan for [PR #15251](https://github.com/netbox-community/netbox/pull/15251))
- I'm not convinced that this is in fact a bug: Rendering a config IMO should be considered a "write" operation as we're sending data to the server for processing (even if it's not stored in the DB)
- I do think we should introduce a separate permission action for it (e.g. `device_renderconfig`)
- ↳ This is a little bit inconsistent between the API and the GUI
- ↳ _Action:_ Jeremy to write up approach to add a permission which we evaluate for both GUI and API. This would be a breaking change.

_[Dan] (Brought forward) Device uniqueness and renaming devices (Re: [#14044](https://github.com/netbox-community/netbox/issues/14044))_
- Dan: IMO we shouldn't do any changes with this model
- Jeremy: Bulk renaming unnamed devices isn't something we can do with the current bulk-renaming code, because an unnamed device's name has no portions to be found-and-replaced, and device names have a uniqueness constraint within each site
- Jeff: Are they looking for a templating functionality like what's offered for creating interfaces and other device components?
- ↳ Jeremy: I don't think that's what they're describing, although that would be a valid FR.
- Dan / Jeremy: _Decision_:
- ↳ Catch the `AttributeError` when devices are at separate sites
- ↳ Catch the `IntegrityError` when devices are at same site
- ↳ Support renaming unnamed devices to a single name, using regex `^$`, as long as they're in separate sites
- ↳ Presumably, display a friendly error if the user tries to bulk-rename a set of devices in the same site

_[Dan] (Brought forward) [#15016](https://github.com/netbox-community/netbox/issues/15016) quick double check on direction_
- Jeremy just needs to find time to dedicate to this one

_[Dan] (Brought forward) Discussion re: allow plugins to exempt specific views from `LOGIN_REQUIRED`_
- Jeremy: I've created and opened a PR for [Replace AUTH_EXEMPT_PATHS setting with a view-specific control #16580](https://github.com/netbox-community/netbox/issues/16580) which obsoletes this one

_[Dan] (Brought forward) Discussion re: allowing plugins to have “base” URLs (`/pnp` instead of `/plugins/pnp`)_
- Jeremy: We reserve the root namespace for core things; a plugin registering a base of e.g. `/dcim` would break things
- Dan: This is for Cisco PNP, the switch boots and makes a request to a URL that always starts with `/pnp`
- Jeremy: Recommend just rewriting the request from the HTTP server handling the frontend
- ↳ Dan: I may have tried that and failed, but I'll take another run at it.

**Comment by FliesLikeABrick** [comment]
Thank you for posting these notes, it may seem minor but it is interesting to see the discourse behind the scenes of a project like this
